### ISSUE TITLE
Receiving cURL Error 92 hitting any endpoint

### ISSUE THREAD

**Issue by jamacon36** [agilecrm/php-api#21]
Hey, thanks for the API!
I'm getting an error using the hitting any endpoint with curl and catching it reveals it to be error 92:
`"HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)"`

My implementation:
```
<?php
$user = get_user($id);        
$agile_data = array(
  'tags' => array('Web User'),
  'properties' => array(
       array(// First Name
         'name' => 'first_name',
          'value' => $user['first_name'],
          'type' => 'SYSTEM'
       ),
       array(// Last Name
         'name' => 'last_name',
         'value' => $user['last_name'],
         'type' => 'SYSTEM'
       ),
       array(// Email
         'name' => 'email',
         'value' => $user['email'],
         'type' => 'SYSTEM'
       ),
       // ETC...
  )
);
$agile_json = json_encode($agile_data);
curl_wrap('contacts', $agile_json, 'POST', 'application/json');
```

Here's my server setup:
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

Apache 2.4
config
config-runtime
mod_bwlimited
mod_cgi
mod_deflate
mod_expires
mod_headers
mod_mpm_prefork
mod_proxy
mod_proxy_fcgi
mod_proxy_http
mod_proxy_wstunnel
mod_ruid2
mod_security2
mod_ssl
mod_unique_id
tools

PHP 7.2
libc-client
pear
php-bcmath
php-calendar
php-cli
php-common
php-curl
php-devel
php-fpm
php-ftp
php-gd
php-imap
php-litespeed
php-mbstring
php-mysqlnd
php-pdo
php-posix
php-sockets
php-xml
runtime
```
I'm able to log in and hit the endpoint just fine from a browser using the same creds. Any help would be appreciated!

Thanks again!

**Comment by aditron** [comment]
I'm getting the same issue too...

**Comment by jamacon36** [comment]
I found the issue @aditron. In the curl_wrapper function the `CURLOPT_HTTPHEADER` `setopt` is improperly formatted out of the box. Removing spaces from it got it working for me:

As it is in the repo:
```
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
    "Content-type : $content_type;", 'Accept : application/json'
));
```
What got it working:
```
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
    "Content-Type:$content_type;", 'Accept:application/json'
));
```

**Comment by aditron** [comment]
Nice one @jamacon36 ! Interesting that it stopped working recently after years of being OK - I suspect a PHP version change or corresponding cURL version change may have been a contributor. Mind you, the complete silence from AgileCRM on the issue wasn't helpful...

**Comment by agilecrm** [comment]
@jamacon36 @aditron We somehow missed this message. However, we have already made the required changes and this should be working fine. 
@jamacon36 Appreciate your time and effort identifying this issue.

**Comment by jamacon36** [comment]
Thanks, @agilecrm ! Would you mind merging PR #22 even though the change got made independently? It'd be cool to get the contribution recorded.

**Comment by rkuntoro** [comment]
We came across this problem too when testing with a PHP7. Fixed it by switching to HTTP 1.1

`curl_setopt($ch, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);`

It's interesting that removing spaces to the header helps too. I'm guessing this is caused by an older server that the API uses.
